api/model/src/testUriCompliance/java/org/projectnessie/model/TestContentKeyEncoding.java

```diff
@@ -0,0 +1,270 @@
+        .containsExactlyInAnyOrderElementsOf(expectedViolations);
+    soft.assertThat(uri.getPath().split("/")[2]).describedAs(uri.getPath()).isEqualTo(uriEncoded);
+
+    uri = HttpURI.from(format("http://hostname/foo/%s/bar", uriEncoded));
+    soft.assertThat(uri.getViolations())
+        .describedAs("path-middle: %s", uriEncoded)
+        .containsExactlyInAnyOrderElementsOf(expectedViolations);
+    soft.assertThat(uri.getPath().split("/")[2]).describedAs(uri.getPath()).isEqualTo(uriEncoded);
+
+    uri = HttpURI.from(format("http://hostname/bar?foo=%s", uriEncoded));
+    soft.assertThat(uri.getViolations()).describedAs("query-param1: %s", uriEncoded).isEmpty();
+    soft.assertThat(uri.getQuery().split("&")[0].split("=")[1])
+        .describedAs(uri.getQuery())
+        .isEqualTo(uriEncoded);
+
+    uri = HttpURI.from(format("http://hostname/bar?foo=%s&bar=moo", uriEncoded));
+    soft.assertThat(uri.getViolations()).describedAs("query-param2: %s", uriEncoded).isEmpty();
+    soft.assertThat(uri.getQuery().split("&")[0].split("=")[1])
+        .describedAs(uri.getQuery())
+        .isEqualTo(uriEncoded);
+  }
+
+  static Stream<Arguments> fromPathString() {
+    return Stream.of(
+        arguments("hello", List.of("hello"), Set.of()),
+        arguments("he#llo", List.of("he#llo"), Set.of()),
+        arguments("he..^llo.world", List.of("he/llo", "world"), Set.of()),
+        arguments("hello..*", List.of("hello."), Set.of()),
+        arguments("..*hello..*", List.of(".hello."), Set.of()),
+        arguments("..*hello..*~", List.of(".hello.~"), Set.of()),
+        arguments("..*hello..*~hello", List.of(".hello.~hello"), Set.of()),
+        arguments("..*hello...*~hello", List.of(".hello", ".~hello"), Set.of()),
+        arguments("..*hello..*.~hello", List.of(".hello.", "~hello"), Set.of()),
+        arguments("..*hello..*...*~hello", List.of(".hello.", ".~hello"), Set.of()),
+        arguments("..*hello..*~.~hello", List.of(".hello.~", "~hello"), Set.of()),
+        arguments("..*~hello..*", List.of(".~hello."), Set.of()),
+        arguments("hello.world", List.of("hello", "world"), Set.of()),
+        arguments("hello..*.world", List.of("hello.", "world"), Set.of()),
+        arguments("hello...*world", List.of("hello", ".world"), Set.of()),
+        arguments("he..*llo.wo..*rld", List.of("he.llo", "wo.rld"), Set.of()),
+        // path separators cause violations
+        arguments("he/llo", List.of("he/llo"), Set.of(AMBIGUOUS_PATH_SEPARATOR)),
+        // control-characters cause violations
+        arguments(
+            "he\u001dllo.wo\u001drld",
+            List.of("he.llo", "wo.rld"),
+            Set.of(SUSPICIOUS_PATH_CHARACTERS)),
+        // backslash is a violation
+        arguments("he\\llo.world", List.of("he\\llo", "world"), Set.of(SUSPICIOUS_PATH_CHARACTERS)),
+        // % alone is ambiguous
+        arguments("he%llo.world", List.of("he%llo", "world"), Set.of(AMBIGUOUS_PATH_ENCODING)));
```

Comment:
WDYT about adding test cases for escaped `%` and `\` chars (no violations)?

Reply:
This test isn't about testing violations - it's about testing `fromPathString` and asserting that the new mechanism doesn't cause violations. There's no URL escaping happening here.